feat(abstract-utxo): add address creation for descriptor wallets

ericli-bitgo opened the pull request:
<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->

ericli-bitgo committed:
chore(abstract-utxo): add wasm-miniscript

TICKET: BTC-1454

ericli-bitgo committed:
feat(abstract-utxo): add address creation for descriptor wallets

TICKET: BTC-1454

ericli-bitgo committed:
test(bitgo): add tests for address creation for descriptor wallets

TICKET: BTC-1454

OttoAllmendinger commented on modules/abstract-utxo/src/abstractUtxoCoin.ts:
```diff
@@ -231,26 +233,48 @@ export interface ParsedTransaction<TNumber extends number | bigint = number> ext
+};
+
+export interface GenerateDescriptorAddressOptions extends GenerateAddressOptions {
+  descriptor: NamedDescriptor;
```
```suggestion
  descriptors: NamedDescriptor[];
```

OttoAllmendinger commented on modules/abstract-utxo/src/abstractUtxoCoin.ts:
```diff
@@ -1052,6 +1084,12 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
     return [KeyIndices.USER, KeyIndices.BACKUP, KeyIndices.BITGO];
   }
 
+  isDescriptorWallet(
+    params: GenerateFixedAddressOptions | GenerateDescriptorAddressOptions | VerifyAddressOptions
+  ): params is GenerateDescriptorAddressOptions {
+    return 'descriptor' in params;
+  }
+
```
this should be a static method

OttoAllmendinger commented on modules/abstract-utxo/src/abstractUtxoCoin.ts:
```diff
@@ -997,23 +1021,31 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
+    let expectedAddress: AddressDetails;
+    if (params.coinSpecific && 'descriptor' in params.coinSpecific) {
+      expectedAddress = this.generateAddress({
+        descriptor: params.coinSpecific.descriptor,
```
we only submit the name of the descriptor (the full descriptor itself is defined on the wallet)

pranavjain97 reviewed (commented):
WP changes look fine. Please request re-review when Otto's changes are addressed

ericli-bitgo committed:
refactor(abstract-utxo): apply code review

- Change VerifyAddressOptions to VerifyAddressOptions<UtxoCoinSpecific>
- Rename to GenerateFixedScriptAddressOptions
- Make isDescriptorWallet static

TICKET: BTC-1454
